uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/PopupMediaItem.h

```diff
@@ -35,24 +34,23 @@ namespace Toolkit
 class PopupMediaItem : public QObject
 {
   Q_OBJECT
-  Q_PROPERTY(QString title READ title NOTIFY popopMediaItemChanged)
-  Q_PROPERTY(QUrl sourceUrl READ sourceUrl NOTIFY popopMediaItemChanged)
-  Q_PROPERTY(QUrl linkUrl READ linkUrl NOTIFY popopMediaItemChanged)
-  Q_PROPERTY(QString caption READ caption NOTIFY popopMediaItemChanged)
-  Q_PROPERTY(PopupMediaType popupMediaType READ popupMediaType NOTIFY popopMediaItemChanged)
+  Q_PROPERTY(QString title READ title NOTIFY popupMediaItemChanged)
+  Q_PROPERTY(QString caption READ caption NOTIFY popupMediaItemChanged)
+  Q_PROPERTY(PopupMediaType popupMediaType READ popupMediaType NOTIFY popupMediaItemChanged)
 
 public:
   PopupMediaItem(QObject* parent = nullptr) = delete;
   explicit PopupMediaItem(PopupMedia* popupMedia, QObject* parent = nullptr);
 
   QString title() const;
-  QUrl linkUrl() const;
-  QUrl sourceUrl() const;
   QString caption() const;
   PopupMediaType popupMediaType() const;
+  PopupMedia* popupMediaItem() const;
+
+  QColor jsonColorHelper(const QJsonArray& colorArray, const int index) const;
```

Comment:
I would consider making this static. It doesn't need anything from the class itself, so it shouldn't be required to have an object to use it.

Also, it's pretty clear already, but `jsonColorHelper` isn't as descriptive as it could be. Maybe something like `toColor` or `extractColor`? Your name is not terrible, so I'm ok if you prefer to keep it.

Reply:
@JamesMBallard  `jsonColorExtractor`?

Reply:
How about `extractColorFromJson`? `jsonColorExtractor` sounds more like the name of a class/utility than the method itself. I find it useful to use verbs for the method names.

Like this:
```c++
struct DoesAThing
{
  void doTheThing();
};

DoesAThing helperThatDoesAThing;
helperThatDoesAThing.doTheThing();
```